<launch>
  <arg name="left_motor_id" doc="path to the bagfile whose messages should be reordered" />
  <arg name="right_motor_id" doc="path to the bagfile whose messages should be reordered" />
  <arg name="fpga_id" doc="fpga_id" />
  <arg name="sample_rate" default="!" doc="sample rate in Hz" />
  <arg name="Kp" default="!" doc="kp" />
  <arg name="Ki" default="!" doc="ki" />
  <arg name="Kd" default="!" doc="kd" />
  <arg name="max_disp" default="!" doc="max_disp" />
  <arg name="min_disp" default="!" doc="min_disp" />
  <arg name="max_steering_angle" default="!" doc="max_steering angle in degrees" />
  <arg name="zero_angle_raw" default="!" doc="zero_angle" />

  <arg name="rate_resolved" value="$(eval '' if arg('sample_rate')=='!' else '--sample_rate ' + arg('sample_rate'))" />
  <arg name="kp_resolved" value="$(eval '' if arg('kp')=='!' else '--kp ' + arg('kp'))" />
  <arg name="ki_resolved" value="$(eval '' if arg('ki')=='!' else '--ki ' + arg('ki'))" />
  <arg name="kd_resolved" value="$(eval '' if arg('kd')=='!' else '--kd ' + arg('kd'))" />
  <arg name="max_disp_resolved" value="$(eval '' if arg('max_disp')=='!' else '--max_disp ' + arg('max_disp'))" />
  <arg name="min_disp_resolved" value="$(eval '' if arg('min_disp')=='!' else '--min_disp ' + arg('min_disp'))" />
  <arg name="angle_resolved" value="$(eval '' if arg('max_steering_angle')=='!' else '--max_steering_angle ' + arg('max_steering_angle'))" />
  <arg name="zero_raw_resolved" value="$(eval '' if arg('zero_angle_raw')=='!' else '--zero_angle_raw ' + arg('zero_angle_raw'))" />

  <node name="steering_controller" pkg="roboy_navigation"
        type="steering_controller.py"
        args="--left_motor_id $(arg left_motor_id) --right_motor_id $(arg right_motor_id) --fpga_id $(arg fpga_id) $(arg rate_resolved) $(arg kp_resolved) $(arg ki_resolved) $(arg kd_resolved) $(arg max_disp_resolved) $(arg min_disp_resolved) $(arg angle_resolved) $(arg zero_raw_resolved)" 
        output="screen"/>

</launch>
